packages/grid/x-data-grid-pro/src/hooks/features/export/serializer/excelSerializer.ts

```diff
@@ -0,0 +1,177 @@
+import type * as Excel from 'exceljs';
+import {
+  GridStateColDef,
+  GridRowId,
+  GridColDef,
+  GridValueOptionsParams,
+  GridValueFormatterParams,
+  GridApi,
+} from '@mui/x-data-grid';
+import { buildWarning } from '@mui/x-data-grid/internals';
+import { GridExceljsProcessInput } from '../gridExcelExportInterface';
+
+const getExcelJs = () => import('exceljs');
+
+const warnInvalidFormattedValue = buildWarning([
+  'MUI: When the value of a field is an object or a `renderCell` is provided, the Excel export might not display the value correctly.',
+  'You can provide a `valueFormatter` with a string representation to be used.',
+]);
+
+const getFormattedValueOptions = (
+  colDef: GridColDef,
+  { id, row, field }: GridValueOptionsParams,
+  api: GridApi,
+) => {
+  if (!colDef.valueOptions) {
+    return [];
+  }
+  let valueOptionsFormatted;
+  if (typeof colDef.valueOptions === 'function') {
+    valueOptionsFormatted = colDef.valueOptions({ id, row, field });
+  } else {
+    valueOptionsFormatted = colDef.valueOptions;
+  }
+
+  if (colDef.valueFormatter) {
+    valueOptionsFormatted = valueOptionsFormatted.map((option) => {
+      if (typeof option === 'object') {
+        return option;
+      }
+
+      const params: GridValueFormatterParams = { field, api, value: option };
+      return String(colDef.valueFormatter!(params));
+    });
+  }
+  return valueOptionsFormatted.map((option) =>
+    typeof option === 'object' ? option.label : option,
+  );
+};
+
+const serializeRow = (id: GridRowId, columns: GridStateColDef[], api: GridApi) => {
+  const row = {};
+  const dataValidation = {};
+
+  const firstCellParams = api.getCellParams(id, columns[0].field);
+  const outlineLevel = firstCellParams.rowNode.depth;
+
+  columns.forEach((column) => {
+    const cellParams = api.getCellParams(id, column.field);
+    switch (cellParams.colDef.type) {
+      case 'singleSelect': {
+        const valueOptions: GridValueOptionsParams = {
+          id: cellParams.id,
+          row: cellParams.row,
+          field: cellParams.field,
+        };
+        const formattedValueOptions = getFormattedValueOptions(
+          cellParams.colDef,
+          valueOptions,
+          api,
+        );
+        dataValidation[column.field] = {
+          type: 'list',
+          allowBlank: true,
+          formulae: [`"${formattedValueOptions.map((x) => x.toString()).join(',')}"`],
+        };
+        const formattedValue = api.getCellParams(id, column.field).formattedValue;
+        if (process.env.NODE_ENV !== 'production') {
+          if (String(cellParams.formattedValue) === '[object Object]') {
+            warnInvalidFormattedValue();
+          }
+        }
+        row[column.field] =
+          typeof formattedValue === 'object' ? formattedValue.label : formattedValue;
```

Comment:
Same thing?

```suggestion
        row[column.field] = formattedValue.label ?? formattedValue;
```

Reply:
Not if `formattedValue` is `undefined`. Another solution is:
```
row[column.field] = formattedValue && (formattedValue.label ?? formattedValue)
```

Reply:
I'm not following. If `formattedValue=undefined` then `row[column.field]` will be `undefined` too, like the previous code.

Reply:
You can use `?` to tell that `formattedValue` may be `undefined`.

```tsx
row[column.field] = formattedValue?.label ?? formattedValue;
```